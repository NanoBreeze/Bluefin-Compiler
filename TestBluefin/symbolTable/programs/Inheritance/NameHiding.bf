/*
If derived class has member fields with same name as base class, we expect derived fields to hide base class' 
*/

struct Base {
	int a;
	bool b;
	
	float f() {}
};

struct Der extends Base {
	bool a;
	Base b;
	
	void f() {} // since we're just testing resolution and declarations, we can only distinguish Der::f() from Base::f() based on return type, not number of params
};

int main() {
	Der der;
	der.f(); 
	der.a;
	der.b;
}