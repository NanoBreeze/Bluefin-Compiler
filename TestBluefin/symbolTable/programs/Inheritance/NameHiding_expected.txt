declare - Base - c_struct - t_Base
setCurrentScope - Level 1
declare - a - c_var - t_int
declare - b - c_var - t_bool
declare - f - c_function - t_float
enterScope - Level 2
exitScope - Level 2
exitScope - Level 1
resolve - Base - c_struct - t_Base
declare - Der - c_struct - t_Der
setCurrentScope - Level 1
declare - a - c_var - t_bool
declare - b - c_var - t_Base
declare - f - c_function - t_void
enterScope - Level 2
resolve - Base::f - c_function - t_float
exitScope - Level 2
exitScope - Level 1
declare - main - c_function - t_int
enterScope - Level 1
declare - der - c_var - t_Der
exitScope - Level 1
resolve - int - c_builtinType - t_int
resolve - bool - c_builtinType - t_bool
resolve - float - c_builtinType - t_float
resolve - bool - c_builtinType - t_bool
resolve - Base - c_struct - t_Base
resolve - void - c_builtinType - t_void
resolve - int - c_builtinType - t_int
resolve - Der - c_struct - t_Der
resolve - der - c_var - t_Der
resolve - Der::f - c_function - t_void
resolve - der - c_var - t_Der
resolve - Der::a - c_var - t_bool
resolve - der - c_var - t_Der
resolve - Der::b - c_var - t_Base
